### PR TITLE
Fixes Issue 161

### LINKED ISSUE #161: Create Archive Feature for User's Note in the Desktop Application
Original Issue #139

**Description**
**As** a user,
**I need** the ability to archieve a note, including the uploaded media file (video, audio, photo) and its associated annotation, 
**So that** I can manage my content by removing unnecessary or outdated uploads.

**Tasks**
- [x] Ensure the archive button triggers the proper function when clicked.
    - [x] Change the "delete" button to "archive" button
- [x] Implement the backend logic to handle the archive of the media file and its associated annotation.
    - [x] Ensure that the note is archived from the database and any associated storage. 
- [x] Create confirmation prompts to avoid accidental deletion. (also note delete is not working can only handle archieve atm)
- [x] Ensure the archive process updates the UI to reflect the removal of the note.
    - [x]  Refresh or remove the archived note from the view without requiring a page reload.
- [x] Verify that the archive is regression-free.

**Acceptance Criteria**
- [ ] Code is well-commented.
- [ ] Code should pass the test case before merging.
- [ ] Users can successfully delete a note from their view.
- [ ] Deleted notes do not reappear after a page refresh or re-login.
- [ ] Backend data is correctly updated to reflect the archive note.
- [ ] Confirmation prompts work as expected before archive.

This issue complete last item on the MVP of the project.

### PR DESCRIPTION
**Pull request by jwang-101**
REFERENCE:
Fixes #161

WHAT CHANGED:
Added functionality to ensure archived notes are filtered out from the map view and side bar, as well as in the user's list of notes. Users can only view non-archived notes, and archived notes will remain hidden as intended

WHY CHANGED:
This was changed because archived notes were previously still visible on the map. This causes confusion and clutter on the user interface if there is irrelevant or outdated pins. This change improves the user experience by only allowing the user to see active notes. 

HOW CHANGED:
Updated the fetchNotes function to exclude archived notes by adding a filter condition (note.isArchived). Personal and global notes are filtered to only include notes that don't have the isArchived flag set to true. 
Then, the map rendering and user note list will also remove notes with the isArchived property, insuring that archived notes remain invisible across all relevant locations.

### CODE REVIEW

File: components/ui/archive.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    </div>
+  );
+};
+
+NoteEditor.propTypes = {
+  note: PropTypes.object.isRequired,
+  isNewNote: PropTypes.bool.isRequired,
+};
+
+export default NoteEditor;
```
Comment (amychen108) [inline components/ui/archive.tsx:63]: Code looks good but does it function as intended?

File: components/ui/archive.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import ApiService from '../../utils/api_service'; // Assuming the correct path
+import PropTypes from 'prop-types';
```
Comment (amychen108) [inline components/ui/archive.tsx:3]: Make sure you import the note page you're using since the delete button is on it.

**Review by amychen108 — CHANGES_REQUESTED** [review]

File: components/ui/archive.tsx
Comment (kate-holdener) [inline components/ui/archive.tsx:1]: Looks like @amychen108 tests are for app/lib/components/noteElements/note_editor and @jwang-101 was supposed to build this component. I am not seeing the node_editor component.

File: components/ui/archive.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    setShowConfirmation(true);
+  };
+
+  const handleCancelClick = () => {
+    setShowConfirmation(false);
+  };
+
+  const handleConfirmClick = async () => {
+    try {
+      const response = await ApiService.archiveNote(note.id);
```
Comment (kate-holdener) [inline components/ui/archive.tsx:20]: I don't see archiveNote in ApiService.

**Review by BaloneyBoy97 — APPROVED** [review]
Good Job Justin, your changes passed all test cases! nice work this week!

**Review by BaloneyBoy97 — APPROVED** [review]
Good Job!

**Review by kate-holdener — CHANGES_REQUESTED** [review]
Please remove commented out code.

File: app/lib/components/noteElements/note_handler.ts [hunk 4]
```diff
@@ -54,37 +54,85 @@ export const handleEditorChange = (
   setEditorContent(content);
 };
 
-export const handleDeleteNote = async (
+//
```
Comment (amychen108) [inline app/lib/components/noteElements/note_handler.ts:57]: Is this needed or no? Not sure since it's commented out

**Review by amychen108 — COMMENTED** [review]
All comments seem good and help explain the code you wrote.
Nice job changing all the deletes into archives instead since we can only archive at the moment.
Functionality is still a work in progress but so far it seems to follow the correct implementation steps you mentioned in your PR.

**Review by BaloneyBoy97 — APPROVED** [review]
great work! all bug fixed! notes are archived correctly!

**Review by yashb196 — APPROVED** [review]

**Pull request merged**